homeassistant/components/juicenet/sensor.py

```diff
@@ -17,104 +23,115 @@
+    ),
+)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the JuiceNet Sensors."""
-    entities = []
     juicenet_data = hass.data[DOMAIN][config_entry.entry_id]
     api = juicenet_data[JUICENET_API]
     coordinator = juicenet_data[JUICENET_COORDINATOR]
 
-    for device in api.devices:
-        for sensor in SENSOR_TYPES:
-            entities.append(JuiceNetSensorDevice(device, sensor, coordinator))
+    entities = [
+        JuiceNetSensorDevice(device, coordinator, description)
+        for device in api.devices
+        for description in SENSOR_TYPES
+    ]
     async_add_entities(entities)
 
 
 class JuiceNetSensorDevice(JuiceNetDevice, SensorEntity):
     """Implementation of a JuiceNet sensor."""
 
-    def __init__(self, device, sensor_type, coordinator):
+    def __init__(self, device, coordinator, description: SensorEntityDescription):
         """Initialise the sensor."""
-        super().__init__(device, sensor_type, coordinator)
-        self._name = SENSOR_TYPES[sensor_type][0]
-        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
-        self._attr_device_class = SENSOR_TYPES[sensor_type][2]
-        self._attr_state_class = SENSOR_TYPES[sensor_type][3]
-
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return f"{self.device.name} {self._name}"
+        super().__init__(device, description.key, coordinator)
+        self.entity_description = description
+        self._attr_name = f"{self.device.name} {description.name}"
 
     @property
     def icon(self):
         """Return the icon of the sensor."""
         icon = None
-        if self.type == "status":
+        if self.entity_description.key == "status":
             status = self.device.status
             if status == "standby":
                 icon = "mdi:power-plug-off"
             elif status == "plugged":
                 icon = "mdi:power-plug"
             elif status == "charging":
                 icon = "mdi:battery-positive"
-        elif self.type == "temperature":
-            icon = "mdi:thermometer"
-        elif self.type == "voltage":
-            icon = "mdi:flash"
-        elif self.type == "amps":
-            icon = "mdi:flash"
-        elif self.type == "watts":
-            icon = "mdi:flash"
-        elif self.type == "charge_time":
-            icon = "mdi:timer-outline"
-        elif self.type == "energy_added":
-            icon = "mdi:flash"
+        else:
+            icon = self.entity_description.icon
         return icon
 
-    @property
-    def unit_of_measurement(self):
-        """Return the unit the value is expressed in."""
-        return self._unit_of_measurement
-
     @property
     def state(self):
         """Return the state."""
         state = None
-        if self.type == "status":
+        sensor_type = self.entity_description.key
+        if sensor_type == "status":
             state = self.device.status
```

Comment:
Side note: Looks like we can simplify by using `getattr` and remove the if/else block.
```py
return getattr(self.device, self.entity_description.key, None)
```